templates/sample-app/res/src/java/main/java/com/financialforce/routesolver/QuestionConsumer.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2017, FinancialForce.com, inc
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ *   are permitted provided that the following conditions are met:
+ *
+ * - Redistributions of source code must retain the above copyright notice,
+ *      this list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ *      this list of conditions and the following disclaimer in the documentation
+ *      and/or other materials provided with the distribution.
+ * - Neither the name of the FinancialForce.com, inc nor the names of its contributors
+ *      may be used to endorse or promote products derived from this software without
+ *      specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ *  OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ *  THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ *  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ *  OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ *  OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **/
+
+package com.financialforce.routesolver;
+
+import com.rabbitmq.client.Channel;
+
+import com.financialforce.orizuru.exception.consumer.handler.HandleMessageException;
+import com.financialforce.orizuru.message.Context;
+import com.financialforce.orizuru.transport.rabbitmq.DefaultConsumer;
+import com.financialforce.routesolver.exception.NoSolverException;
+import com.financialforce.routesolver.interfaces.ISolver;
+import com.financialforce.routesolver.problem.avro.Answer;
+import com.financialforce.routesolver.problem.avro.Question;
+import com.financialforce.routesolver.solver.Solver;
+
+/**
+ * Handles Vehicle Routing Questions and provides a mechanism to solve these problems.
+ * <p>
+ * Extension of {@link DefaultConsumer}.
+ */
+public class QuestionConsumer extends DefaultConsumer<Question, Answer> {
+
+	private ISolver<Question, Answer> solver;
+
+	public QuestionConsumer(Channel channel) {
+		super(channel, Question.class.getName(), Answer.class.getName());
+		this.solver = new Solver();
+	}
+
+	@Override
+	public Answer handleMessage(Context context, Question question) throws HandleMessageException {
+
+		if (solver == null) {
```

Comment:
The only constructor sets the solver. Mark it as final, then you won't have to validate it's not null here.